ext/date/php_date.c

```diff
@@ -3726,12 +3726,70 @@ PHP_METHOD(DateTimeImmutable, setTimestamp)
 }
 /* }}} */
 
+/* {{{ */
+PHP_METHOD(DateTimeImmutable, setMicroseconds)
+{
+	zval *object, new_object;
+	php_date_obj *dateobj, *new_dateobj;
+	zend_long us;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "l", &us) == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	if (UNEXPECTED(us < 0 || us > 999999)) {
+		zend_throw_error(date_ce_date_range_error,
+						"Microseconds must be between 0 and 999999, "ZEND_LONG_FMT" given",
+						us);
+		RETURN_THROWS();
```

Comment:
I tend to use this formatting for multi-line calls:

```
		zend_throw_error(
			date_ce_date_range_error,
			"Microseconds must be between 0 and 999999, " ZEND_LONG_FMT " given",
			us
		);
```

I'd be nice if you could that as well.